modules/gke-nodepool/main.tf

```diff
@@ -164,7 +164,28 @@ resource "google_container_node_pool" "nodepool" {
       content {
         count              = var.node_config.guest_accelerator.count
         type               = var.node_config.guest_accelerator.type
-        gpu_partition_size = var.node_config.guest_accelerator.gpu_partition_size
+        gpu_partition_size = var.node_config.guest_accelerator.gpu_driver == null ? null : var.node_config.guest_accelerator.gpu_driver.partition_size
+
+        dynamic "gpu_sharing_config" {
+          for_each = var.node_config.guest_accelerator.gpu_driver != null ? [""] : []
+          content {
+            gpu_sharing_strategy       = var.node_config.guest_accelerator.gpu_driver.max_shared_clients_per_gpu != null ? "TIME_SHARING" : null
+            max_shared_clients_per_gpu = var.node_config.guest_accelerator.gpu_driver.max_shared_clients_per_gpu
+          }
+        }
+
+        dynamic "gpu_driver_installation_config" {
+          for_each = var.node_config.guest_accelerator.gpu_driver != null ? [""] : []
+          content {
+            gpu_driver_version = var.node_config.guest_accelerator.gpu_driver.version
+          }
+        }
+      }
+    }
+    dynamic "local_nvme_ssd_block_config" {
+      for_each = var.node_config.local_nvme_ssd_count != null ? [""] : []
```

Comment:
Would this work if `local_nvme_ssd_count` is 0?

If it doesn't you can probably use a condition like this `coalesce(var.node_config.local_nvme_ssd_count, 0) > 0`